### ISSUE TITLE
length of output string of Tdi internal DATE_TIME()

### ISSUE THREAD

**Issue by zack-vii** [MDSplus/mdsplus#500]
when calling DATE_TIME() via mdsip the length of the string is not correct. This is given by the fact that the mdsip server is not reading the right length. This is provoked by the DATE_TIME internals:
LibSysAscTim() allocates 24 chars (zero terminated) but DATE_TIME accepts only 23 chars.
The zero-termination ('\0') is not StrCopyR-copied to str. However, DATE_TIME will expand out_ptr to len=24 chars appending a random character due to the memory breach.
Suggestion:
1.) LibSysAscTim() should not zero terminate in the if(time_str) case and only allocate 23 chars.(cleaner solution, does it strictly zero terminate in the else case?)
2.) DATE_TIME should only set the new length if  len < (out_ptr->pointer->length)

``` tdi
DATE_TIME(0Xafd58ae0a0c69dQU)
"18-SEP-2015 11:08:57.52"
mdsconnect('myserver')
mdsvalue('DATE_TIME(0Xafd58ae0a0c69dQU)')
"18-SEP-2015 11:08:57.52?"   # ? is a random char
```

**Comment by tfredian** [comment]
What version of MDSplus is running on the server you are connecting to? There was a bug in the alpha branch which could cause this behavior but that should have been fixed in version 7.0.238.

**Comment by zack-vii** [comment]
you are right, sorry. it is fixed in the newest version.
